config/op-consumer/public/delta-sync-dispatching.js

```diff
@@ -64,31 +65,40 @@ async function dispatch(lib, data) {
     if (originalDeletes.length) {
       // Map deletes from OP to DL model
       const transformedDeletes = await transformStatements(fetch, deletesWithContext);
+      const filteredDeletes = removeContextTriples(transformedDeletes, originalDeletes, deletesWithContext);
 
-      if (!transformedDeletes.length) {
+      if (!filteredDeletes.length) {
         console.log(`Warn: Delete statements mapped to empty result.`);
         console.log(`Input: ${deletesWithContext}`);
-        console.log(`Output: ${transformedDeletes}`);
+        console.log(`Output: ${filteredDeletes}`);
       } else {
-        await deleteFromTargetGraph(lib, transformedDeletes);
+        await deleteFromTargetGraph(lib, filteredDeletes);
       }
     }
 
     if (originalInserts.length) {
       // Map inserts from OP to DL model
       const transformedInserts = await transformStatements(fetch, insertsWithContext);
+      const filteredInserts = removeContextTriples(transformedInserts, originalInserts, insertsWithContext);
 
-      if (!transformedInserts.length) {
+      if (!filteredInserts.length) {
         console.log(`Warn: Insert statements mapped to empty result.`);
         console.log(`Input: ${insertsWithContext}`);
-        console.log(`Output: ${transformedInserts}`);
+        console.log(`Output: ${filteredInserts}`);
       } else {
-        await insertIntoTargetGraph(lib, transformedInserts);
+        await insertIntoTargetGraph(lib, filteredInserts);
       }
     }
   }
 }
 
+// Function to serialize object for comparison
+function serializeTriple({ graph, subject, predicate, object }) {
```

Comment:
Not sure what the input parameters are; can't estimate here how robust the serialization would be. Ideally we use some battle tested lib; maybe considering this one here too:
https://github.com/lblod/delta-consumer/blob/master/lib/utils.js#L67

Maybe what you made just works, just want to double check